dockets/queue.py

```diff
@@ -93,11 +93,38 @@ def item_key(self, item):
                             for key_component in self.key)
         return str(item)
 
+    def move_delayed_item(self):
+        pipeline = self.redis.pipeline()
+
+        # Move any due delayed items to the main working list
+        try:
+            pipeline.watch(self._delayed_queue_key())
+            try:
+                next_envelope_key, timestamp = pipeline.zrange(
+                    self._delayed_queue_key(), 0, 1, withscores=True)[0]
+                if time.time() < timestamp:
+                    # It's not yet that time.
+                    return
+                envelope = pipeline.hget(self._payload_key(), next_envelope_key)
+                pipeline.multi()
+                if self.mode == self.FIFO:
+                    pipeline.lpush(self._queue_key(), envelope)
+                else:
+                    pipeline.rpush(self._queue_key(), envelope)
+                pipeline.zrem(self._delayed_queue_key(), next_envelope_key)
+                pipeline.hdel(self._payload_key(), next_envelope_key)
+                pipeline.execute()
+
+            except IndexError:
```

Comment:
This would probably look better if we checked to see if there was actually anything in the list returned by `zrange`. If you're absolutely opposed to doing that, at least put the `except` closer to the list access so it doesn't swallow stuff you didn't intend.